### ISSUE TITLE
RichTextField with no block elements renders with unhelpful UI components

### ISSUE THREAD

**Issue by dkirkham** [wagtail/wagtail#9706]
### RichTextField with no block elements renders with unhelpful UI components

This is more a cosmetic issue, than a bug per se. It might be more of a bug from an accessibility point of view.

When specifying a RichTextField _without_ any block elements in the feature list, the widget still renders:
* an empty menu of block elements when the circle/plus icon is clicked
* the placeholder text suggesting the / key be used to get a list of enabled to block elements, and then an empty menu of block elements.

Both have the potentially confusing "No results" text in the menu because there are no block element types to list there.

### Steps to Reproduce

1. Start a new project
2. Edit the HomePage model to include the field `text = RichTextField(features=['bold', 'italic', 'link'])`
3. Makemigrations/migrate/runserver etc.
4. Navigate to the rich text field in the page editor
5. Observe the placeholder text
<img width="512" alt="image" src="https://user-images.githubusercontent.com/1977376/203310111-35699425-e08c-4489-a439-0743c26793d7.png">

6. Observe the menu when / is typed
<img width="491" alt="image" src="https://user-images.githubusercontent.com/1977376/203310173-738de260-66aa-4189-9581-097ba87ce07d.png">

7. Observe the menu when the circle/plus icon is pressed
<img width="493" alt="image" src="https://user-images.githubusercontent.com/1977376/203310252-94bcdde7-c0fb-4df4-8bcd-084cafb3a6ac.png">

I consider this a bug, because when there are no block elements, the placeholder text, / and circle/plus elements should not be present. This requirement is very similar to the goals of #8249, however in my use case I still want to allow paragraphs and inline formatting; I just don't want headers, lists, etc.

-   I have confirmed that this issue can be reproduced as described on a fresh Wagtail project: (yes)

### Technical details

-   Python version: 3.8.
-   Django version: 4.1.
-   Wagtail version: 4.1.1. Suspect all versions >=4.0 are affected
-   Browser version: Firefox 103, Chrome 105.0.5195.125.

**Comment by thibaudcolas** [comment]
Thank you for the report @dkirkham! This feels like a clear UX issue. I’m not too clear on what the right fix or fixes would be for this, and whether it qualifies as a bug or not.

I agree with your plan on how to fix this, though am finding this a bit hard to reason about so any further input / feedback is very welcome.

**Comment by dkirkham** [comment]
@thibaudcolas, is the issue my use case (ie. not wanting the block elements) being unreasonable, or is it deciding exactly how to present the UI in this circumstance?

Since the issue doesn’t prevent any work being done, nor does it corrupt any data or cause a crash, I’m quite happy for this not to be classified as a bug, but instead as a feature request to clean up the UI.

**Comment by nicklee** [comment]
Definitely an interesting one, so this happens when there aren't any blocks avaliable, but the user can select things like bold/italic/link via the floating formatting bar, which is shown if they select or double click on the text they've entered.

My initial thoughts are that if there are no actions to be complete, then the user shouldn't be presented with the plus symbol. Currently this just results in a wasted click, and potential confusion due to the "no results" text. 

The slight downside of removing the plus icon, will be that if users are familiar with seeing it, they then maybe confused as to why it isn't showing.

With this in mind I feel like updating the placeholder text to simply "Write something" instead of "Write something or type "/" to insert a block" when no blocks are avaliable, as well as not showing the plus symbol, would make it obvious to these users that this is intended.

So to summarise, my suggestion would be for us to do the following when no blocks are avaliable:

- Update placeholder text to "Write something"
- Hide the plus symbol that appears to the left of the form field

**Comment by nicklee** [comment]
It seems we're agreeing that this would be a good solution. I've attached a mockup of how this would look:

![page-editor_no-blocks](https://user-images.githubusercontent.com/298766/211330928-d322624c-df8a-4bfb-b68d-44298a82e920.png)

**Comment by nicklee** [comment]
This has been approved to be implemented in the future

**Comment by th3hamm0r** [comment]
+1
We hardly ever use plain `TextField`s, instead we use `RichtTextField`s with limited features, so having no blocks is quite common for us, which also resulted in some bug reports on our side since we've upgraded to wagtail versions with the new richtext UI.

**Comment by shayan-oanda** [comment]
Hi all, any progress on this?

**Comment by thibaudcolas** [comment]
I’d like to see this happen but have other priorities for the time being. @shayan-oanda if you want to make a pull request please go for it.

**Comment by Zogsha** [comment]
Hello, I tried to solve this issue, but I have absolutely no idea how to do a pull request and contribute (I have no programming background)

I used the gitpod-wagtail-develop.
I just modified the `getSharedPropsFromOptions` in `wagtail/client/src/components/Draftail/index.js` and set `placeholder`, `topToolbar `and `commandToolbar` with a condition on `blockTypes.length` :

```javascript
// wagtail/client/src/components/Draftail/index.js - line 209
return {
// ...
placeholder: blockTypes.length > 0 ? gettext('Write something or type ‘/’ to insert a block') : gettext('Write something'),
// ...
topToolbar: (props) => (
        <>
          {(() => {
            if (blockTypes.length > 0) {
              return <BlockToolbar
                {...props}
                triggerIcon={ADD_ICON}
                triggerLabel={comboBoxTriggerLabel}
                comboLabel={comboBoxLabel}
                comboPlaceholder={comboBoxLabel}
                noResultsText={gettext('No results !!!')}
                ComboBoxComponent={ComboBox}
              />;
            } else {
              return null;
            }
          })()}
          <InlineToolbar
            {...props}
            pinButton={pinButton}
            defaultToolbar={getSavedToolbar()}
            onSetToolbar={onSetToolbar}
          />
        </>
      ),
// ...
commandToolbar: (props) => (
        <>
          {(() => {
            if (blockTypes.length > 0) {
              return <CommandPalette
                {...props}
                noResultsText={gettext('No results !!!')}
                ComboBoxComponent={ComboBox}
              />;
            } else {
              return null;
            }
          })()}
        </>
      ),
// ...
};
```

To demonstrate, I modified a `TextField ` in the bakerydemo into a `RichTextFIeld` with only inline styles. As you can see below, it looks like it works just fine.

![recording-2023-11-21-14-15-22](https://github.com/wagtail/wagtail/assets/17270547/86ab5a45-51c3-44f0-9fdd-c845e077e1e0)

What do you think ?
